core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -158,10 +158,20 @@ class PartKeyLuceneIndex(dataset: Dataset,
   def partIdsEndedBefore(endedBefore: Long): EWAHCompressedBitmap = {
     val collector = new PartIdCollector()
     val deleteQuery = LongPoint.newRangeQuery(PartKeyLuceneIndex.END_TIME, 0, endedBefore)
-    searcherManager.acquire().search(deleteQuery, collector)
+
+    withNewSearcher(s => s.search(deleteQuery, collector))
     collector.result
   }
 
+  private def withNewSearcher(func: IndexSearcher => Unit): Unit = {
```

Comment:
Just one suggestion: you could make `withNewSearcher` return a value to make code slightly better:

    def withNewSearcher[B](func: IndexSearcher => B): B